Automattic/wp-calypso
# Pass the siteSlug or siteId to further steps in the hero flow to support refreshing

#### Changes proposed in this Pull Request

Fixes https://github.com/Automattic/wp-calypso/issues/58339 By passing the `siteSlug` or `siteId` which is a required dependency for all steps of the hero flow.

Also passes the query param when users use the back button.

#### Testing instructions
Navigate to 
`/start/setup-site/intent?flags=signup/hero-flow&siteSlug=YOUR_SITE.wordpress.com` or
`/start/setup-site/intent?flags=signup/hero-flow&siteId=$YOUR_SITE_ID`

Navigate through the hero flow and assert that the siteId or siteSlug is passed to the next step.
Use browser refresh to refresh the page and verify that it loads correctly.
